src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -231,43 +329,23 @@ def dtype(self) -> DtypeObj:
         >>> idx.dtype
         dtype('int64')
         """
+        # TODO: SNOW-1458123 implement dtype
         self.to_pandas_warning()
         return self.to_pandas().dtype
 
-    def astype(self, dtype: Any, copy: bool = True) -> Index:
+    @property
+    def inferred_type(self) -> None:
```

Comment:
add @index_not_implemented()?

Reply:
This is being used somewhere in the tests

Reply:
but this method has no implementation here

Reply:
I don't remember what test was failing but I'll add the decorator

Reply:
`test_series_setitem_with_empty_key_and_empty_item_negative` fails if it is raising not implemented error